lib/inc/libral/cwrapper.hpp

```diff
@@ -8,7 +8,7 @@ extern "C" {
 
 uint8_t get_providers(char **result);
 uint8_t get_resource(char **result, char *type_name, char *resource_name);
-uint8_t set_resource(char **result, char *type_name, char *resource_name, int valuesc, char **values);
+uint8_t set_resource(char **result, char *type_name, char *resource_name, int desired_attributes_c, char **desired_attributes);
```

Comment:
I just noticed this now, and should be done in a separate PR: all these functions should really have docs/comments on what they do and what they return.